python/pfs/drp/stella/lsf.py

```diff
@@ -0,0 +1,688 @@
+
+        Besides the difference in return types from the ``computeArray``
+        method (this returns a `Kernel1D`, while ``computeArray`` returns a
+        an array), this method provides a model for the LSF centered at zero.
+        This serves as the foundation for the other methods in the class.
+
+        This abstract methods requires definition by subclasses.
+
+        Parameters
+        ----------
+        center : `float`, pixels
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        kernel : `Kernel1D`
+            Kernel with a model of the LSF at the nominated position.
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+    def computePeak(self, point=None):
+        """Return the value of the peak pixel in the LSF image
+
+        Parameters
+        ----------
+        point : `lsst.geom.Point2D`, optional
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        value : `float`
+            Value of the peak pixel in the LSF image.
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        return self.computeKernel(point.getX())[0]
+
+    def computeShape(self, point=None):
+        """Return the shape of the Lsf as if it's a ``Psf``
+
+        This method is included for compatibility with `lsst.afw.detection.Psf`.
+
+        Parameters
+        ----------
+        point : `lsst.geom.Point2D`, optional
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        shape : `lsst.afw.geom.ellipses.Quadrupole`
+            The second moments of the Lsf.
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        sigma = self.computeShape1D(point.getX())
+        return Quadrupole(sigma**2, 0, 0)
+
+    def computeShape1D(self, center):
+        """Return the standard deviation at the nominated position
+
+        The standard deviation is a measure of the width of the LSF.
+
+        Parameters
+        ----------
+        center : `float`, pixels
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        stdev : `float`
+            Standard deviation of the LSF at the nominated position.
+        """
+        return self.computeKernel(center).computeStdev()
+
+    def computeApertureFlux(self, radius, point=None):
+        """Return the flux within the nominated radius
+
+        Parameters
+        ----------
+        point : `lsst.geom.Point2D`, optional
+            Position at which to evaluate the LSF.
+
+        Returns
+        -------
+        flux : `float`
+            Flux within the nominated radius.
+        """
+        if point is None:
+            point = self.getAveragePosition()
+        kernel = self.computeKernel(point.getX())
+        intRadius = int(radius)
+
+        flux = trapz([kernel[ii] for ii in range(-intRadius, intRadius + 1)])
```

Comment:
How do you know that a trapezoidal integration is good enough?  Does the (much smarter but sampling-sensitive) SDSS code run?  If the LSF is oversampled (as in general it will be) we need to run on the fully-sampled version not on this instrument-sampled version